src/psyclone/domain/lfric/transformations/lfric_alg_invoke_2_psy_call_trans.py

```diff
@@ -58,16 +67,323 @@ def validate(self, node, options=None):
+            name="inc_x_plus_by")
+        builtin_metadata["inc_x_plus_by"] = metadata
+        # inc_x_minus_by metadata
+        metadata = LFRicKernelMetadata(
+            operates_on="dof",
+            meta_args=[FieldArgMetadata(
+                "gh_real", "gh_readwrite", "any_space_1"),
+                       ScalarArgMetadata("gh_real", "gh_read"),
+                       FieldArgMetadata("gh_real", "gh_read", "any_space_1")],
+            procedure_name="inc_x_minus_by_code",
+            name="inc_x_minus_by")
+        builtin_metadata["inc_x_minus_by"] = metadata
+        # x_divideby_y metadata
+        metadata = LFRicKernelMetadata(
+            operates_on="dof",
+            meta_args=[FieldArgMetadata("gh_real", "gh_write", "any_space_1"),
+                       FieldArgMetadata("gh_real", "gh_read", "any_space_1"),
+                       FieldArgMetadata("gh_real", "gh_read", "any_space_1")],
+            procedure_name="x_divideby_y_code",
+            name="x_divideby_y")
+        builtin_metadata["x_divideby_y"] = metadata
+        # inc_x_divideby_y metadata
+        metadata = LFRicKernelMetadata(
+            operates_on="dof",
+            meta_args=[FieldArgMetadata(
+                "gh_real", "gh_readwrite", "any_space_1"),
+                       FieldArgMetadata("gh_real", "gh_read", "any_space_1")],
+            procedure_name="inc_x_divideby_y_code",
+            name="inc_x_divideby_y")
+        builtin_metadata["inc_x_divideby_y"] = metadata
+        # a_times_x metadata
+        metadata = LFRicKernelMetadata(
+            operates_on="dof",
+            meta_args=[FieldArgMetadata("gh_real", "gh_write", "any_space_1"),
+                       ScalarArgMetadata("gh_real", "gh_read"),
+                       FieldArgMetadata("gh_real", "gh_read", "any_space_1")],
+            procedure_name="a_times_x_code",
+            name="a_times_x")
+        builtin_metadata["a_times_x"] = metadata
+        # inc_a_times_x metadata
+        metadata = LFRicKernelMetadata(
+            operates_on="dof",
+            meta_args=[ScalarArgMetadata("gh_real", "gh_read"),
+                       FieldArgMetadata(
+                           "gh_real", "gh_readwrite", "any_space_1")],
+            procedure_name="inc_a_times_x_code",
+            name="inc_a_times_x")
+        builtin_metadata["inc_a_times_x"] = metadata
+
+        for kern_call in node.children:
+            if isinstance(kern_call, LFRicBuiltinFunctor):
+                kernel_metadata = builtin_metadata[kern_call.name.lower()]
+            else:
+                kernel = kernels[id(kern_call)]
+                kernel_metadata = self._get_metadata(kernel)
+            arg_idx = 0
+            for meta_arg in kernel_metadata.meta_args:
+                if not check_args:
+                    arg = kern_call.children[arg_idx]
+                    self._add_arg(arg, arguments)
+                # If this is a stencil arg then skip any additional
```

Comment:
Missing a `continue`? (Otherwise I don't understand why there's `if not check_args:`)

Reply:
I've added words to the docstring to explain. This method has 2 modes. 1) checking the arguments and 2) creating the argument list. If checking the arguments (check_args=True) we don't want to create the argument list. The main reason for this is that we can get an index error as arg_idx may end up being greater than the number of children if there are fewer algorithm arguments than are expected by the metadata.